File: Linear_Search/PHP/mitogh/linear_search_test.php
<?php
require "linear_search.php";

class linear_search_test extends PHPUnit_Framework_TestCase{
    protected $numbers;

    public function setUp(){
        $this->numbers = [123, 324, 3242, 1, 4, -123, -32, 234, 657, 87, 345, 786, 867, 34564];
    }

    public function test_key_found(){
        $this->assertTrue(linear_search($this->numbers, 1));
        $this->assertTrue(linear_search($this->numbers, 786));
        $this->assertTrue(linear_search($this->numbers, 34564));
        $this->assertTrue(linear_search($this->numbers, 123));
    }

    public function test_key_not_found(){
        $this->assertFalse(linear_search($this->numbers, 0));
        $this->assertFalse(linear_search($this->numbers, 20));
        $this->assertFalse(linear_search($this->numbers, 100));
        $this->assertFalse(linear_search($this->numbers, 9));
    }

    public function test_wrong_arguments(){
        $this->assertFalse(linear_search($this->numbers, true));
        $this->assertFalse(linear_search($this->numbers, false));
        $this->assertFalse(linear_search($this->numbers));
        
        $this->assertFalse(linear_search());
    }
}

File: Linear_Search/Java/mitogh/test/LinearSearchTest.java
import org.junit.Test;
import static org.junit.Assert.*;

public class LinearSearchTest {
    LinearSearch linearSearch;
    
    public LinearSearchTest() {
        linearSearch = new LinearSearch();
    }
    
    @Test
    public void searchIntKeyTest() {
        int numbers[] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
        System.out.println("Search for keys that exist in the array.");
        assertTrue(linearSearch.search(numbers, 9));
        assertTrue(linearSearch.search(numbers, 1));
        assertTrue(linearSearch.search(numbers, 5));
        assertTrue(linearSearch.search(numbers, 10));
        assertTrue(linearSearch.search(numbers, 4));
        
        System.out.println("Search for keys that do not exist in the array.");
        assertFalse(linearSearch.search(numbers, 0));
        assertFalse(linearSearch.search(numbers, 20));
        assertFalse(linearSearch.search(numbers, 100));
        assertFalse(linearSearch.search(numbers, -70));
        assertFalse(linearSearch.search(numbers, 11));
    }    
}
